sky/server/server.py

```diff
@@ -267,6 +273,112 @@ async def dispatch(self, request: fastapi.Request, call_next):
         return await call_next(request)
 
 
+class BearerTokenMiddleware(starlette.middleware.base.BaseHTTPMiddleware):
+    """Middleware to handle Bearer Token Auth (Service Accounts)."""
+
+    async def dispatch(self, request: fastapi.Request, call_next):
+        # Only process requests with the a path prefix /sa
+        if not request.url.path.startswith('/sa/'):
+            return await call_next(request)
+
+        # Check if service account tokens are enabled
+        sa_enabled = os.environ.get(constants.ENV_VAR_ENABLE_SERVICE_ACCOUNTS,
```

Comment:
Wondering if we should make this a configuration item instead of env var

Reply:
Since this requires additional setup in the deployment, making it a config may be quite confusing, as setting the config to True in dashboard does not mean it is turned on.